src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1742,6 +1741,11 @@ internal static bool IsCaptured(Symbol variable, SourceMethodSymbol containingSy
                     throw ExceptionUtilities.UnexpectedValue(variable.Kind);
             }
 
+            if (containingSymbol.GetIsNewExtensionMember() && ReferenceEquals(variable.ContainingSymbol, containingSymbol.ContainingSymbol))
```

Comment:
>ReferenceEquals(variable.ContainingSymbol, containingSymbol.ContainingSymbol) [](http://example.com/codeflow?start=62&length=77)

Is this an indirect/veiled way to check whether the ```variable``` is an extension parameter? Consider checking for that explicitly instead.